pom.xml

```diff
@@ -205,4 +211,20 @@
             </dependency>
         </dependencies>
     </dependencyManagement>
+    <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-surefire-plugin</artifactId>
+                    <configuration>
+                        <systemPropertyVariables>
+                            <java.util.logging.config.file>src/test/resources/logging.properties</java.util.logging.config.file>
+                            <org.jvnet.hudson.test.HudsonTestCase.slaveDebugPort>${slaveDebugPort}</org.jvnet.hudson.test.HudsonTestCase.slaveDebugPort>
```

Comment:
```suggestion
```

Reply:
without those option how can I activate this when running tests from a console?

Reply:
Is this for setting breakpoints in code run inside an agent JVM? If so, that should be automatic I think when you run Maven in the debugger. If that is not happening, then it is something that can be improved in `jenkins-test-harness`. I suspect you rediscovered https://github.com/jenkinsci/jenkins-test-harness/pull/428#discussion_r863031976? At any rate, do not commit stuff like this to plugin POMs. Either fix JTH, or temporarily edit your POM or whatever script you use to run tests in a debugger, but do not include the edit in commits (I have come to grumblingly accept `git add -p` as a necessity).

Reply:
using something such ` mvn test -Dtest=JUnitResultArchiverTest#slave -Dmaven.surefire.debug` doesn't start the slave in debug mode.
upgrading to last `1752.v86627a_c48d91` I went from 
```
$ "/Library/Java/JavaVirtualMachines/temurin-11.jdk/Contents/Home/bin/java"  -Djava.awt.headless=true -Xmx512M -jar "/Volumes/workspace/dev/sources/jenkins/junit-plugin/target/jenkins-for-test/WEB-INF/lib/remoting-4.12.jar"
```

to 
```
$ "/Library/Java/JavaVirtualMachines/temurin-11.jdk/Contents/Home/bin/java"  -Djava.awt.headless=true -Xmx512m -XX:+PrintCommandLineFlags -jar "/Volumes/workspace/dev/sources/jenkins/junit-plugin/target/jenkins-for-test/WEB-INF/lib/remoting-4.12.jar"
-XX:G1ConcRefinementThreads=13 -XX:GCDrainStackTargetSize=64 -XX:InitialHeapSize=536870912 -XX:MaxHeapSize=536870912 -XX:+PrintCommandLineFlags -XX:ReservedCodeCacheSize=251658240 -XX:+SegmentedCodeCache -XX:+UseCompressedClassPointers -XX:+UseCompressedOops -XX:+UseG1GC 
```
not sure what is the need of those per default extra jvm setup? 

> At any rate, do not commit stuff like this to plugin POMs. 

why? what is the technical problem here? there is a default which do not activate anything.
using an option such if '-Dmaven.surefire.debug` then automatically run the slave in debug. why?
with such configuration it's possible to default what is happening on the unit but not on the slave or activate both or activate only debug in the slave.

Reply:
> why?

Because debugging setup has nothing to do with your change. If such POM configuration is needed—which I doubt—it should be added to `plugin-pom`, not a particular plugin. If you need to temporarily `stash` some edits to help develop a change, fine, but do not commit them.

Most likely what is needed is a simple change to JTH to automatically activate the debugger on agent JVMs when the test case is being debugged, which is something that already happens for the controller implicitly in `JenkinsRule` but explicitly in `RealJenkinsRule`.

Reply:
> > why?
> 
> Because debugging setup has nothing to do with your change. If such POM configuration is needed—which I doubt—it should be added to `plugin-pom`, not a particular plugin. If you need to temporarily `stash` some edits to help develop a change, fine, but do not commit them.

I removed because it looks forbidden even if there is no other way to do it...


> 
> Most likely what is needed is a simple change to JTH to automatically activate the debugger on agent JVMs when the test case is being debugged, which is something that already happens for the controller implicitly in `JenkinsRule` but explicitly in `RealJenkinsRule`.